programs/sbf/build.rs

```diff
@@ -37,6 +37,11 @@ fn rerun_if_changed(files: &[&str], directories: &[&str], excludes: &[&str]) {
 }
 
 fn main() {
+    if env::var("CI_TEST_CHECKS_FORCE_HOST_COMPILE").is_ok() {
+        println!("cargo:warning=(not a warning) Compiling with host toolchain for CI...");
+        return;
+    }
+
```

Comment:
@alessandrod this is a bit aggressive, but i think it's better not to use `cargo-build-sbf` altogether here.

- Faster build (no need to wget/tar, etc; also build cache reuse..)
- Avoid broken `clippy` with `programs/sbf/build.rs`
- nicer output in general.
- don't pull in externally-controlled dep (i think latest `solana-sbf-tools` is implicitly used?)

potential downsides:

- version mismatch between monorepo rust vs  `solana-sbf-tools` => shouldn't cause too much trouble because casually monorepo is newer
- potential risk of super hard-to-debug bugs due to this special cased `build.rs` behavior => possible, so I'm adding `cargo:warning=...` here for this bailout codepath. 

if this makes sense, I'll finish up this pr in this direction (squash commits, toss review to @yihau, etc).

Reply:
hope this is what you wanted (faster ci; a car), not faster horse. lol

Reply:
haha this looks good to me, although not sure if it impacts how @dmakarov tests his changes

Reply:
as discussed at discord, we've reached conclution this is ok given that this is new additional fail-first test step: https://discord.com/channels/428295358100013066/560503042458517505/1082870294504538244

![image](https://user-images.githubusercontent.com/117807/223614380-3e3c72ec-659e-4cff-91e8-3c535d600c51.png)